t/unit/tasks/test_tasks.py

```diff
@@ -983,6 +983,17 @@ def test_regular_task(self):
                 name='George Costanza', test_eta=True, test_expires=True,
             )
 
+            # With ETA, absolute expires in the past in ISO format.
+            presult2 = self.mytask.apply_async(
+                kwargs={'name': 'George Costanza'},
+                eta=self.now() + timedelta(days=1),
+                expires=(self.now() - timedelta(days=2)).isoformat(),
+            )
+            self.assert_next_task_data_equal(
+                consumer, presult2, self.mytask.name,
+                name='George Costanza', test_eta=True, test_expires=True,
+            )
```

Comment:
@auvipy the test is here. 

I was looking for better solution to test it. Because this doesn't explain why expires is in string format when task is retried. The reason is that with apply async task request is serialized into queue and retry use serialized data (expires is one of them). To be honest I don't like the different behavior of apply (eager mode) and apply async. This creates a lot of bugs.

Reply:
this is pull request with the bug https://github.com/celery/celery/pull/6957

Reply:
can you recheck tests?

Reply:
yes I am working on it. But if I fix pylint/pep8 isort check fails.

Reply:
yeah you are right. can you try to increase code coverage a bit?